jdeeco-core/src/cz/cuni/mff/d3s/deeco/runtimelog/RuntimeLogger.java

```diff
@@ -0,0 +1,537 @@
+	 * This {@link Writer} serves for writing the time offsets for backward jumps
+	 * in the main log file.
+	 * @see RuntimeLogger 
+	 */
+	private BufferedWriter backLogWriter;
+
+	/**
+	 * The size of the data (number of bytes) so far being written
+	 * into the {@link RuntimeLogger#dataWriter}. 
+	 */
+	private long currentDataOffset;
+	/**
+	 * There are {@link SnapshotProvider}s registered on the {@link RuntimeLogger}
+	 * being held in this collection before the {@link RuntimeLogger#init} method is called.
+	 * With each {@link SnapshotProvider} there is associated period, specifying,
+	 * how often the {@link SnapshotProvider} should be called.
+	 * After the {@link RuntimeLogger#init} method is called the stored
+	 * {@link SnapshotProvider}s are processed and this collection is cleared.
+	 */
+	private final Map<SnapshotProvider, Long> snapshotProviders;
+	/**
+	 * There are the time offsets registered on the {@link RuntimeLogger}
+	 * being held in this collection before the {@link RuntimeLogger#init} method is called.
+	 * After the {@link RuntimeLogger#init} method is called the stored time offsets
+	 * are written using the {@link RuntimeLogger#backLogWriter} and this collection
+	 * is cleared.
+	 */
+	private final List<Long> backLogOffsets;
+	/**
+	 * The {@link CurrentTimeProvider} that is used to add time value into each record
+	 * being written using the {@link RuntimeLogger#dataWriter} or {@link RuntimeLogger#indexWriter}. 
+	 */
+	private CurrentTimeProvider timeProvider;
+	/**
+	 * The {@link Scheduler} that is used to plan the {@link Task}s
+	 * that are responsible for periodic invocation of the registered {@link SnapshotProvider}s.
+	 */
+	private Scheduler scheduler;
+
+	/**
+	 * The singleton instance of the {@link RuntimeLogger}. This singleton holds the
+	 * {@link Writer}s that access the log files. Multiple access to these files is not
+	 * desirable.
+	 */
+	private static RuntimeLogger instance = new RuntimeLogger();
+
+	/**
+	 * Specifies the default file where the logging records of the runtime events are written.
+	 * This file destination can be overridden using the
+	 * {@link RuntimeLogger#init(CurrentTimeProvider, Scheduler, Writer, Writer, Writer)}
+	 * method.
+	 */
+	private static final File DATA_FILE = new File("logs/runtimeData.xml");
+	/**
+	 * Specifies the default file where the index of the runtime log is stored.
+	 * This file destination can be overridden using the
+	 * {@link RuntimeLogger#init(CurrentTimeProvider, Scheduler, Writer, Writer, Writer)}
+	 * method.
+	 */
+	private static final File DATA_INDEX_FILE = new File("logs/dataIndex.xml");
+	/**
+	 * Specifies the default file where the back log time offsets are written.
+	 * This file destination can be overridden using the
+	 * {@link RuntimeLogger#init(CurrentTimeProvider, Scheduler, Writer, Writer, Writer)}
+	 * method.
+	 */
+	private static final File BACK_LOG_OFFSET_TABLE_FILE = new File("logs/backLogOffsetTable.xml");
+	
+	/**
```

Comment:
Does it really make sense to append? I would say no. A better option would be to generate a new file like:
runtimeData-001.xml
runtimeData-002.xml etc.
However, I would wait with this numbering feature. In any case, I would remove appending. 

Much better feature would be to be able to configure the log directory in runtime configuration (speak with Ilias).